datafusion/common/src/column.rs

```diff
@@ -372,7 +372,7 @@ impl FromStr for Column {
 
 impl fmt::Display for Column {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.flat_name())
+        write!(f, "{}", self.quoted_flat_name())
```

Comment:
I thought identifier are the one double quoted in your ticket 😕, but it seems you are modifying column

Reply:
identifier could includes dot, that is the reason we need double quotation to differentiate it from qualified column, it is also relatively rare compare to column, so we don't have double quote every where.

In duckdb, they also has double quote for alias name (identifier).

```
D select t.a from t;
┌───────┐
│   a   │
│ int32 │
├───────┤
│     1 │
└───────┘
D select t.a as t.a from t;
Parser Error: syntax error at or near "."
LINE 1: select t.a as t.a from t;
                       ^
D select t.a as "t.a" from t;
┌───────┐
│  t.a  │
│ int32 │
├───────┤
│     1 │
└───────┘
```

If we want to introduce dot notation for struct `a.b`, it is consistent to duckdb if we don't have double quote for column.

https://duckdb.org/docs/sql/data_types/struct.html

Reply:
> I think only adding double quote for identifier that includes dot is probably a better idea.
> 
> If we have column that has name includes dot, we display it like `qualifier."name.with.dot"`

I'm rethinking what you said. I did a test for the function name using the original method (flat_name).
```
        let schema = Schema::new(vec![
            Field::new("id", DataType::Int32, false),
            Field::new("MAX(id)", DataType::Int32, false),
            Field::new("state", DataType::Utf8, false),
        ]);

        let plan = table_scan(Some("t"), &schema, None)
            .unwrap()
            .filter(col("state").eq(lit("CO")))
            .unwrap()
            .project(vec![col("MAX(id)"), max(col("id"))])
            .unwrap()
            .build()
            .unwrap();

-----
 right: "Projection: t.MAX(id), MAX(t.id)\n  Filter: t.state = Utf8(\"CO\")\n    TableScan: t"
```
We can distinguish between them easily.
Indeed, maybe we don't need to quote the identifier if it contains a bracket. Only quoting it for the dot might be simpler. 🤔